File: get_indexed_pages.py
import requests
from bs4 import BeautifulSoup
import json

def handler(request):
    domain = request.args.get("domain")
    if not domain:
        return {"statusCode": 400, "body": json.dumps({"error": "Missing domain"})}

    headers = {"User-Agent": "Mozilla/5.0"}
    search_url = f"https://www.google.com/search?q=site:{domain}"
    html = requests.get(search_url, headers=headers).text
    soup = BeautifulSoup(html, "html.parser")
    stats = soup.select_one("#result-stats")

    if stats:
        text = stats.text
        return {"statusCode": 200, "body": json.dumps({"indexed_pages": text})}
    else:
        return {"statusCode": 200, "body": json.dumps({"indexed_pages": "Not found"})}

File: get_indexed_pages.js
const fetch = require('node-fetch');
const cheerio = require('cheerio');

module.exports = async (req, res) => {
  const { domain } = req.query;

  if (!domain) {
    return res.status(400).json({ error: "Missing domain parameter" });
  }

  const headers = { "User-Agent": "Mozilla/5.0" };
  const searchUrl = \`https://www.google.com/search?q=site:\${domain}\`;

  try {
    const response = await fetch(searchUrl, { headers });
    const html = await response.text();
    const $ = cheerio.load(html);
    const stats = $('#result-stats').text() || 'Not found';
    res.status(200).json({ indexed_pages: stats });
  } catch (error) {
    res.status(500).json({ error: "Failed to fetch indexed pages", details: error.toString() });
  }
};
